concurrent_swiss_map.go

```diff
@@ -27,6 +27,30 @@ type shard[K comparable, V any] struct {
 	*sync.RWMutex
 }
 
+// OptFunc is a type that is used in New function for passing options.
+type OptFunc[K comparable, V any] func(o *CsMap[K, V])
+
+// New function creates *CsMap[K, V].
+func New[K comparable, V any](options ...OptFunc[K, V]) *CsMap[K, V] {
+	m := CsMap[K, V]{
```

Comment:
We may create private a generator function that we can use both in New and Create, what do you think @gozeloglu ?

Reply:
Yes, I had this idea, but the `Create` function will be removed soon. That's why the private generator function will be used only for the `New` function and I did not move the code into a helper private function.